packages/core/src/fhir-deduplication/resources/observation-shared.ts

```diff
@@ -0,0 +1,118 @@
+import { CodeableConcept, Observation } from "@medplum/fhirtypes";
+import { LOINC_CODE, LOINC_OID, SNOMED_CODE, SNOMED_OID } from "../../util/constants";
+
+export const observationStatus = [
+  "entered-in-error",
+  "unknown",
+  "registered",
+  "preliminary",
+  "final",
+  "amended",
+  "corrected",
+  "cancelled",
+] as const;
+
+export type ObservationStatus = (typeof observationStatus)[number];
+
+export const statusRanking = {
```

Comment:
Can we set the type of this as `Record<ObservationStatus, number>`? We should try to use the compiler to help us avoid a mismatch between the statuses and the ranking.